### DISCUSSION TITLE
Cargo missions going to wrong factions / Missionevents need "requireddestinationfaction"

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#15170] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Events can check if there are specific adjacent locations (requiredDestinationTypes), but not specific adjacent factions.

This leads to illogical scenarios where a Coalition station can ask you to transport things to a Separatist station, so the mission text has to be kept as vague as possible. 

You can specify the faction in the eventset, but it doesn't actually seem to do anything regarding the DESTINATION of the mission.
**Nothing in defined in the mission, missionevent, or eventset that enforces a specific faction as the destination**

![image](https://github.com/user-attachments/assets/c9897ad5-fc5f-4830-9ffc-9b960d061a51)
![image](https://github.com/user-attachments/assets/6eaa6e18-98d4-433b-980c-6efb99365910)

In vanilla
![image](https://github.com/user-attachments/assets/8380599c-18ab-457a-a4bb-d2bae1b5957c)

You can potentially do two things:

- Add `requireddestinationfaction` to check if there's an adjacent outpost of the designated faction, that also takes `requiredDestinationTypes` into account. Also add a parameter for "samefaction" that checks the current location's faction as a way to easily set transport missions between say two coalition locations without making coalition variants of the mission 
-  Add "faction" to LocationType in missions **<LocationType from="AnyOutpost" to="AnyOutpost" Faction="Coalition" />**

It's not a very practical solution as there are 3 layers you need to change to make it work (eventset, missionevent, mission) 

### Version

v1.6.19.1 (Unto the Breach Update Hotfix 2)

**Comment by Regalis11** [comment]
This is something we've intentionally kept as-is. Making the cargo mission require an adjacent location that's both of the correct type and owned by the correct faction would mean cargo missions could often not be given at all.
